src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-notification-list/rgw-bucket-notification-list.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, Input, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import { BehaviorSubject, Observable, of } from 'rxjs';
+import { RgwBucketService } from '~/app/shared/api/rgw-bucket.service';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { TableComponent } from '~/app/shared/datatable/table/table.component';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { URLBuilderService } from '~/app/shared/services/url-builder.service';
+import { Bucket } from '../models/rgw-bucket';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+import { catchError, switchMap } from 'rxjs/operators';
+import { TopicConfiguration } from '~/app/shared/models/notification-configuration.model';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+
+const BASE_URL = 'rgw/bucket';
+@Component({
+  selector: 'cd-rgw-bucket-notification-list',
+  templateUrl: './rgw-bucket-notification-list.component.html',
+  styleUrl: './rgw-bucket-notification-list.component.scss',
+  providers: [{ provide: URLBuilderService, useValue: new URLBuilderService(BASE_URL) }]
+})
+export class RgwBucketNotificationListComponent extends ListWithDetails implements OnInit {
+  @Input() bucket: Bucket;
+  @ViewChild(TableComponent, { static: true })
+  table: TableComponent;
+  permission: Permission;
+  tableActions: CdTableAction[];
+  columns: CdTableColumn[] = [];
+  selection: CdTableSelection = new CdTableSelection();
+  notification$: Observable<TopicConfiguration[]>;
+  subject = new BehaviorSubject<TopicConfiguration[]>([]);
+  context: CdTableFetchDataContext;
+  @ViewChild('filterTpl', { static: true })
+  filterTpl: TemplateRef<any>;
+  @ViewChild('eventTpl', { static: true })
+  eventTpl: TemplateRef<any>;
+
+  constructor(
+    private rgwBucketService: RgwBucketService,
+    private authStorageService: AuthStorageService,
+    public actionLabels: ActionLabelsI18n
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    this.permission = this.authStorageService.getPermissions().rgw;
+    this.columns = [
+      {
+        name: $localize`Name`,
+        prop: 'Id',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Topic`,
+        prop: 'Topic',
+        flexGrow: 1
+      },
+      {
+        name: $localize`Event`,
+        prop: 'Event',
+        flexGrow: 1,
+        cellTemplate: this.eventTpl
```

Comment:
you don't need to create a template for it. you can reuse the already existing one.  The only problem is that it is not carbonized yet.. I remember [we had a PR](https://github.com/ceph/ceph/pull/60361) from you for it but we couldn't finish it last time. Maybe we can use the badge here for now and replace it later through the tables component?

thoughts @pujaoshahu @afreen23 
```suggestion
        cellTransformation: CellTemplate.badge,
        customTemplateConfig: {
          class: 'badge-background-primary'
        }
```